addons/maintenance/migrations/13.0.1.0/post-migration.py

```diff
@@ -0,0 +1,22 @@
+from openupgradelib import openupgrade
+
+
+def replace_field_active_for_archive(env):
+    column = openupgrade.get_legacy_name('active')
+    if openupgrade.column_exists(
+            env.cr, "maintenance_request", column):
+        openupgrade.logged_query(
+            env.cr, """
+            UPDATE maintenance_request
+            SET archive =
+            CASE
+            WHEN {column} = true THEN false
+            WHEN {column} = false THEN true
+            END
+            """.format(column=column),
+            )
```

Comment:
Như này chưa ok lắm vì nếu có 1 bản ghi đã archive nhưng chưa inactive, thì chạy qua cái này sẽ thành chưa archive?
Những cái đã archive rồi thì nên giữ nguyên, mình chỉ set lại giá trị trường archive cho những bản ghi chưa archive nhưng có active = False thôi
```suggestion
        openupgrade.logged_query(
            env.cr, """
            UPDATE maintenance_request
            SET archive = TRUE
            WHERE archive = FALSE AND {column} = FALSE
            """.format(column=column),
            )
```

Reply:
xong rồI anh ạ.